recipe-box-frontend/src/main/frontend/src/components/RecipesList/RecipesGrid.tsx

```diff
@@ -25,10 +27,14 @@ export default function RecipesGrid({
         const controller = new AbortController();
         const signal = controller.signal;
 
+        const storedUserId = sessionStorage.getItem("userId");
+        const userId = storedUserId !== null ? parseInt(storedUserId, 10) : undefined;
+
         let url = `${backendUrl}/recipes/${mealType ? `?mealType=${mealType}` : ""}`;
         console.log(mealType);
 
         fetch(
+            privateRecipes ? `${backendUrl}/recipes/search/private?userId=${userId}` : 
```

Comment:
You can try and do this if you have an userId and if the user is a cook/chef, and by doing that you will eliminate the privateRecipes prop

Reply:
A cook can acces both private recipes and public recipes and because of that I can't delete privateRecipes prop, because I can't decide what types of recipes he acces based on the usertype/userId